pkg/ruler/compat.go

```diff
@@ -147,8 +149,14 @@ func MetricsQueryFunc(qf rules.QueryFunc, queries, failedQueries prometheus.Coun
 
 			// Return unwrapped error.
 			return result, origErr
-		}
 
+		} else if err != nil {
+			// When remote querier enabled, only consider failed queries those returning a 500 status code.
+			st, ok := status.FromError(err)
+			if ok && st.Code() == http.StatusInternalServerError {
+				failedQueries.Inc()
+			}
+		}
 		return result, err
```

Comment:
If possible I would prefer to see check for http/grpc errors in `RemoteQuerier`.